packages/react-native-web/src/exports/TextInput/index.js

```diff
@@ -323,18 +330,22 @@ const TextInput: React.AbstractComponent<
   }
 
   function handleSelectionChange(e) {
-    if (onSelectionChange) {
-      try {
-        const node = e.target;
-        const { selectionStart, selectionEnd } = node;
-        e.nativeEvent.selection = {
-          start: selectionStart,
-          end: selectionEnd
-        };
+    try {
+      const node = e.target;
+      const { selectionStart, selectionEnd } = node;
```

Comment:
nitpick: I would do this in a single line

```suggestion
      const { selectionStart, selectionEnd } = e.target;
```